pkg/schema/util.go

```diff
@@ -69,22 +72,83 @@ func (h *Helper) getGoTypeFromSchema(
 	case "integer":
 		return "int64"
 	case "array":
+		itemsSchemaRefName := schema.Items.Ref
+		if itemsSchemaRefName == "" {
+			if schemaRef.Ref != "" {
+				// Here, we deal with a situation like we find in the ECR API
+				// with ImageScanFinding. There is an ImageScanFindings schema
+				// of type "object" which has a "findings" property that is a
+				// reference to an ImageScanFindingsList schema that is of type
+				// "array" of "object" with a set of properties that is
+				// identical to the ImageScanFinding schema:
+				//
+				//     ImageScanFinding:
+				//       properties:
+				//         attributes:
+				//           $ref: '#/components/schemas/AttributeList'
+				//         description:
+				//           $ref: '#/components/schemas/FindingDescription'
+				//         name:
+				//           $ref: '#/components/schemas/FindingName'
+				//         severity:
+				//           $ref: '#/components/schemas/FindingSeverity'
+				//         uri:
+				//           $ref: '#/components/schemas/Url'
+				//       type: object
+				//     ImageScanFindingList:
+				//       items:
+				//         properties:
+				//           attributes:
+				//             $ref: '#/components/schemas/AttributeList'
+				//           description:
+				//             $ref: '#/components/schemas/FindingDescription'
+				//           name:
+				//             $ref: '#/components/schemas/FindingName'
+				//           severity:
+				//             $ref: '#/components/schemas/FindingSeverity'
+				//           uri:
+				//             $ref: '#/components/schemas/Url'
+				//         type: object
+				//       type: array
+				//     ImageScanFindings:
+				//       properties:
+				//         findingSeverityCounts:
+				//           $ref: '#/components/schemas/FindingSeverityCounts'
+				//         findings:
+				//           $ref: '#/components/schemas/ImageScanFindingList'
+				//         imageScanCompletedAt:
+				//           $ref: '#/components/schemas/ScanTimestamp'
+				//         vulnerabilitySourceUpdatedAt:
+				//           $ref: '#/components/schemas/VulnerabilitySourceUpdateTimestamp'
+				//       type: object
+				//
+				// In order to come up with a Go type for the
+				// `ImageScanFindings.Findings` struct field, we need to search
+				// for a schema called ImageScanFindingList, not a schema
+				// called "findings".
+				itemsSchemaRefName = strings.TrimPrefix(schemaRef.Ref, compSchemasRef)
+			} else {
+				itemsSchemaRefName = propNames.Original
+			}
+		} else {
+			itemsSchemaRefName = strings.TrimPrefix(itemsSchemaRefName, compSchemasRef)
+		}
 		itemsSchema := h.getSchemaFromSchemaRef(schema.Items)
 		var itemType string
 		if itemsSchema.Type == "object" {
-			itemType = h.deduceArrayOfObjectsType(schemaName, itemsSchema)
+			itemType = h.deduceArrayOfObjectsType(itemsSchemaRefName, itemsSchema)
 			if itemType == "" {
 				itemType = "!!! UNKNOWN ARRAY OBJECT TYPE !!!"
```

Comment:
Would it be possible to return an error along with the resulting string, and the itemType be empty instead?

Reply:
yeah, totally. I'll fix up.